src/goto-symex/builtin_functions.cpp

```diff
@@ -588,27 +588,33 @@ void goto_symext::symex_cpp_new(const expr2tc &lhs, const sideeffect2t &code)
 
   cur_state->rename(rhs);
   expr2tc rhs_copy(rhs);
+  expr2tc ptr_rhs(rhs);
 
   symex_assign(code_assign2tc(lhs, rhs), true);
 
-  // Mark that object as being dynamic, in the __ESBMC_is_dynamic array
-  type2tc sym_type = array_type2tc(get_bool_type(), expr2tc(), true);
-  expr2tc sym = symbol2tc(sym_type, "__ESBMC_is_dynamic");
-
-  expr2tc ptr_obj = pointer_object2tc(pointer_type2(), lhs);
-  expr2tc idx = index2tc(get_bool_type(), sym, ptr_obj);
-  expr2tc truth = gen_true_expr();
-
-  symex_assign(code_assign2tc(idx, truth), true);
+  expr2tc ptr_obj = pointer_object2tc(pointer_type2(), ptr_rhs);
+  track_new_pointer(ptr_obj, newtype, size);
 
   dynamic_memory.emplace_back(
     rhs_copy, cur_state->guard, false, symbol.name.as_string());
 }
 
-// XXX - implement as a call to free?
-void goto_symext::symex_cpp_delete(const expr2tc &)
+void goto_symext::symex_cpp_delete(const expr2tc &expr)
```

Comment:
Why not implement it as a call to symex_free()?

Reply:
Hi @fbrausse, Do you mean to use `symex_free()` inside this function?

Reply:
We should probably check that the allocation has actually been performed with `new[]` for `delete[]`, but other than that I think symex_cpp_delete() can just delegate to symex_free().